File: JavaPartI/la_longestAscendingSubarray.java
public class Solution {
  public int longest(int[] array) {
    // Write your solution here.
    if (array == null || array.length == 0) {
      return 0;
    }
    int n = array.length, globalMax = 1;
    int[] dp = new int[n];
    dp[0] = 1;
    for (int i = 1; i <= n - 1; i++) {
      if (array[i] > array[i - 1]) {
        dp[i] = dp[i - 1] + 1;
      } else {
        dp[i] = 1;
      }
      if (dp[i] > globalMax) {
        globalMax = dp[i];
      }
    }
    return globalMax;
  }
}

File: pythonPart/la_longestAscendingSubarray.py
class Solution(object):
  def longest(self, array):
    """
    input: int[] array
    return: int
    """
    # write your solution here

    if array is None or len(array) == 0:
      return 0
    n = len(array)
    dp = [1] * n
    maxval = 1
    for i in range(1, n):
      if array[i] > array[i - 1]:
        dp[i] = dp[i - 1] + 1
        maxval = max(maxval, dp[i])
      else:
        dp[i] = 1
    return maxval
